### PR TITLE
Captures anonymous inner class information

### PR DESCRIPTION
**Pull request by jeannielynnmoulton**
~Please review final commit only. This is waiting on https://github.com/diffblue/cbmc/pull/2517 to be merged.~

Test-gen: https://github.com/diffblue/test-gen/pull/2031

### CODE REVIEW

File: jbmc/src/java_bytecode/java_bytecode_parser.cpp [hunk 1]
```diff
@@ -1634,6 +1634,9 @@ void java_bytecode_parsert::rinner_classes_attribute(
     if(!is_inner_class)
       continue;
     parsed_class.is_static_class = is_static;
+    // This is a marker that a class is anonymous.
+    if(inner_name_index == 0)
```
Comment (thk123) [inline jbmc/src/java_bytecode/java_bytecode_parser.cpp:1638]: Nit: Might be simpler to say `parsed_class.is_anonymous_class = inner_name_index == 0;`

**Review by thk123 — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: a29328dbfd3f644f646b284add74a75222fd0cc1).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/78189828

**Comment by thomasspriggs** [comment]
The tests on these three classes in the unit test are not checking the value of `is_anonymous`. Can you add this check to them?
```
  GIVEN(
    "Some package-private class files in the class path with local classes ")
  {
    const symbol_tablet &new_symbol_table = load_java_class(
      "ContainsLocalClass", "./java_bytecode/java_bytecode_parser");
    WHEN("Parsing the InnerClasses attribute for a package-private class ")
    {
      THEN("The class should be marked as package-private")
      {
        const symbolt &class_symbol =
          new_symbol_table.lookup_ref("java::ContainsLocalClass$1LocalClass");
        const java_class_typet java_class =
          to_java_class_type(class_symbol.type);
        REQUIRE(java_class.get_is_inner_class());
        REQUIRE(java_class.get_access() == ID_private);
      }
    }
  }

  const symbol_tablet &new_symbol_table =
    load_java_class("StaticInnerClass", "./java_bytecode/java_bytecode_parser");
  GIVEN("Some class with a static inner class")
  {
    WHEN("Parsing the InnerClasses attribute for a static inner class ")
    {
      THEN("The class should be marked as static")
      {
        const symbolt &class_symbol = new_symbol_table.lookup_ref(
          "java::StaticInnerClass$PublicStaticInnerClass");
        const java_class_typet java_class =
          to_java_class_type(class_symbol.type);
        REQUIRE(java_class.get_is_inner_class());
        REQUIRE(java_class.get_is_static_class());
      }
    }
    WHEN("Parsing the InnerClasses attribute for a non-static inner class ")
    {
      THEN("The class should not be marked as static")
      {
        const symbolt &class_symbol = new_symbol_table.lookup_ref(
          "java::StaticInnerClass$PublicNonStaticInnerClass");
        const java_class_typet java_class =
          to_java_class_type(class_symbol.type);
        REQUIRE(java_class.get_is_inner_class());
        REQUIRE_FALSE(java_class.get_is_static_class());
      }
    }
  }
```

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 06117ae1f9acb76504c1648ef4b6c94f9192b580).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/78290069

**Review by thomasspriggs — APPROVED** [review]
LGTM

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: af0ce5a1925ebe445f39f336baccd190fa39edb1).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/78296851

**Pull request merged**